### ISSUE TITLE
Implicit solve

### ISSUE THREAD

**Issue by alansaillet** [devitocodes/devito#1404]
Hello,

Is it possible for the moment (or will it be possible in the future?) to use implicit scheme to solve equations?
I have tons of ideas to implement with this library, but implicit scheme is a must, in most cases, to ensure the convergence of the model...

Best regards

**Comment by FabioLuporini** [comment]
Hi,

we were talking about this a few days back with @rhodrin . We probably are going to sprint on this soon. Can't give an ETA though, but it's not gonna be before the next couple of months

there are a few escape hatches you could use for the time being though. I'll let @rhodrin comment as he was the one working on them

**Comment by rhodrin** [comment]
Hi @alansaillet,

As @FabioLuporini mentioned this is something we have been discussing and have plans to implement in the fairly near future, although we do not currently have a specific timeline.

In case you're interested in taking a look, @ArthurGiannotta, a Brazilian student, did some work on this and his branch (on a fork) can be found here: https://github.com/ArthurGiannotta/devito. There's also this branch: https://github.com/devitocodes/devito/tree/break_devito I set up for a PhD student who's solving Navier stokes problems. This file may be of interest: https://github.com/devitocodes/devito/blob/break_devito/examples/cfd/lid_driven_cavity.py.

So basically we need to get around to combining and cleaning up these branches and figuring out an appropriate API which I hope should happen soonish.

**Comment by alansaillet** [comment]
Thank's for the information, (and for this awesome library)!
